src/app/inventory/ItemIcon.tsx

```diff
@@ -54,11 +61,27 @@ export function getItemImageStyles(item: DimItem, className?: string) {
  * with respect to masterwork, season icons, mod overlays, etc.
  *
  * This renders just a fragment - it always needs to be rendered inside another div with class "item".
+ *
+ * Since this is used a *lot*, it should not use any hooks, subscriptions, etc.
  */
 export default function ItemIcon({ item, className }: { item: DimItem; className?: string }) {
   const classifiedPlaceholder =
     item.icon === d2MissingIcon && item.classified && getBucketSvgIcon(item.bucket.hash);
   const itemImageStyles = getItemImageStyles(item, className);
+
+  const backgrounds =
+    item.iconDef && itemConstants
+      ? compact([
+          item.tier > 1 && itemConstants.gearTierOverlayImagePaths[item.tier],
+          item.crafted && itemConstants.craftedBackgroundPath,
+          item.iconDef.foreground,
+          // In game the masterwork glow is in front of the icon but I think that looks bad.
+          item.masterwork && itemConstants.masterworkOverlayPath,
+          item.holofoil && itemConstants.holofoil900AnimatedBackgroundOverlayPath,
```

Comment:
Check for item trait "releases.v900.core" (1858131755) for whether to use holofoil900 or old holofoilBackgroundOverlayPath, unless Bungie left another flag for it somewhere

Reply:
(This probably will not stand the test of time after they do another release since that trait won't be on the gun and the release traits aren't sequential), but they haven't given us a way to actually determine which holofoil bg to use yet).